File: algorithms/implementation/strange-advertising/strange-advertising-python/tests/test_viralAdvertising.py
from viralAdvertising import viralAdvertising

class TesteViralAdvertising:
    
    def testeCaso1(self):
        
        n = 3
        esperado = 9
        
        resultado = viralAdvertising(n)
        
        assert resultado == esperado
        
    def testeCaso2(self):
        
        n = 4
        esperado = 15
        
        resultado = viralAdvertising(n)
        
        assert resultado == esperado
        
        
    def testeCaso3(self):
        
        n = 50
        esperado = 2068789129
        
        resultado = viralAdvertising(n)
        
        assert resultado == esperado

File: algorithms/implementation/strange-advertising/strange-advertising-javascript/js/test_viralAdvertising.js
import {viralAdvertising} from '../js/viralAdvertising.js';
import assert from 'assert';

describe('viralAdvertising', function() {

    it('As curtidas acumuladas naquele dia é 9', function() {

        let n = 3;
        let esperado = 9;

        let resultado = viralAdvertising(n);

        assert.strictEqual(resultado, esperado);
    });
});

describe('viralAdvertising', function() {

    it('As curtidas acumuladas naquele dia é 15', function() {

        let n = 4;
        let esperado = 15;

        let resultado = viralAdvertising(n);

        assert.strictEqual(resultado, esperado);
    });
});

describe('viralAdvertising', function() {

    it('As curtidas acumuladas naquele dia é 2068789129', function() {

        let n = 50;
        let esperado = 2068789129;

        let resultado = viralAdvertising(n);

        assert.strictEqual(resultado, esperado);
    });
});